internal/rpc/starknet/types.go

```diff
@@ -52,10 +52,7 @@ func (id *BlockId) UnmarshalJSON(data []byte) error {
 	switch t := token.(type) {
 	case json.Number:
 		value, err := t.Int64()
-		if err != nil {
-			return err
-		}
-		if value < 0 {
+		if err != nil || value < 0 {
```

Comment:
The actual error is getting suppressed here. We need to look into details regarding what type of error could be returned. Also if the value cannot be less than 0 why is `uint64` not being used?

Reply:
`uint64` isn't used because the [`json.Number`](https://pkg.go.dev/encoding/json#Number) only supports `Int64`, `String`, or `Float`. But yes, I could convert the `int64` to `uint64` when stored in `id.value`

Reply:
Done -> https://github.com/NethermindEth/juno/pull/377/commits/fcfaf802764d63ce2cdc6ee870c9ef2ed8c6d8ec